ClientApp/src/app/core/services/auth.service.ts

```diff
@@ -0,0 +1,48 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { JwtHelperService } from '@auth0/angular-jwt';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+@Injectable({
+   providedIn: 'root',
+})
+export class AuthService {
+   private url = 'https://localhost:5001/connect/token';
+   constructor(private http: HttpClient, private jwtHelper: JwtHelperService) {}
+   authenticate(username: string, password: string): Observable<any> {
+      const headers = new HttpHeaders({
+         'Content-Type': 'application/x-www-form-urlencoded',
+      });
+      const body = new URLSearchParams();
+      body.set('username', username);
+      body.set('password', password);
+      body.set('grant_type', 'password');
+      body.set('client_id', 'singer.client');
+      body.set('client_secret', 'secret');
+
+      return this.http
+         .post<any>(this.url, body.toString(), {
+            headers: headers,
+         })
+         .pipe(
+            map(jwt => {
+               if (jwt && jwt.access_token) {
+                  localStorage.setItem('token', JSON.stringify(jwt));
+               }
+            })
+         );
+   }
+
+   isAuthenticated() {
+      const token = localStorage.getItem('token');
```

Comment:
This implies we will always get the token from local storage for every page the user loads. Is that common practice? I would assume this is less performant than reading it from local storage once and keeping it in memory for the duration of the session